docker/docker.go

```diff
@@ -26,11 +26,10 @@ func NewInfoService() *InfoService {
 // GetRuntimeConfigurationFromDockerEngine retrieves information from a Docker environment
 // and returns a map of labels.
 func (service *InfoService) GetRuntimeConfigurationFromDockerEngine() (*agent.RuntimeConfiguration, error) {
-	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion(agent.SupportedDockerAPIVersion))
+	cli, err := getCli()
 	if err != nil {
 		return nil, err
 	}
-	defer cli.Close()
```

Comment:
I don't think that `defer` can be removed, the same applies for the other cases.